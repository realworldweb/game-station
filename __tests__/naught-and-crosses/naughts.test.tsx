import { render, screen, fireEvent } from '@testing-library/react';
import Naughts from '@/pages/naughts-and-crosses';

describe('naughts', () => {
	it('renders naughts unchanged', () => {
		const { container } = render(<Naughts />);

		expect(container).toMatchSnapshot();
	});

	describe('board', () => {
		it('renders board', () => {
			render(<Naughts />);

			expect(screen.getByTitle('game-board')).toBeInTheDocument();
		});

		it('has 9 clickable tiles', () => {
			render(<Naughts />);

			const tiles = screen.getAllByTitle('board-tile');

			expect(tiles).toHaveLength(9);
		});

		it('if x player clicks tile tile should be filled with x ', () => {
			render(<Naughts />);

			const button = screen.getByRole('button', { name: /1 player/i });

			fireEvent.click(button);

			const nameForm = screen.getByRole('form', { name: /player-names/i });

			fireEvent.submit(nameForm);

			const xButton = screen.getByRole('button', { name: /x/i });

			fireEvent.click(xButton);

			const tiles = screen.getAllByTitle('board-tile');

			fireEvent.click(tiles[0]);

			expect(tiles[0]).toHaveTextContent('x');
		});

		it('if o player clicks tile tile should be filled with o ', () => {
			render(<Naughts />);

			const button = screen.getByRole('button', { name: /1 player/i });

			fireEvent.click(button);

			const nameForm = screen.getByRole('form', { name: /player-names/i });

			fireEvent.submit(nameForm);

			const oButton = screen.getByRole('button', { name: /o/i });

			fireEvent.click(oButton);

			const tiles = screen.getAllByTitle('board-tile');

			fireEvent.click(tiles[0]);

			expect(tiles[0]).toHaveTextContent('o');
		});
	});

	it('if player 1 takes turn details should switch if not playing comp', () => {
		render(<Naughts />);

		const button = screen.getByRole('button', { name: /2 player/i });

		fireEvent.click(button);

		const nameForm = screen.getByRole('form', { name: /player-names/i });

		fireEvent.submit(nameForm);

		const oButton = screen.getByRole('button', { name: /o/i });

		fireEvent.click(oButton);

		const tiles = screen.getAllByTitle('board-tile');

		const playerDetails = screen.getByTitle('current-player');

		expect(playerDetails).toHaveTextContent('Player: player1');

		fireEvent.click(tiles[0]);

		expect(playerDetails).toHaveTextContent('Player: player2');
	});
});
