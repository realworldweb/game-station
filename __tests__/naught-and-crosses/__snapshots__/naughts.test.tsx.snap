// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`naughts renders naughts unchanged 1`] = `
<div>
  <div
    class="relative flex grow flex-none flex-wrap w-100 h-14 content-center justify-center bg-slate-600"
  >
    <div
      class="absolute flex flex-col gap-2 bg-white/75 flex-wrap z-10 w-full h-full justify-center content-center "
      title="signs-selection"
    >
      <button
        class="bg-slate-900 text-white px-4 py-2 rounded-md"
      >
        1 player
      </button>
      <button
        class="bg-slate-900 text-white px-4 py-2 rounded-md"
      >
        2 player
      </button>
    </div>
    <p
      class="text-white mb-1"
      title="current-player"
    >
      Player: player1
    </p>
    <div
      class="relative flex border-2 border-white flex-wrap w-96 h-96"
      title="game-board"
    >
      <div
        class="flex flex-wrap justify-center content-center w-1/3 h-1/3 text-white bg-slate-800"
        title="board-tile"
      />
      <div
        class="flex flex-wrap justify-center content-center w-1/3 h-1/3 text-white bg-red-400"
        title="board-tile"
      />
      <div
        class="flex flex-wrap justify-center content-center w-1/3 h-1/3 text-white bg-slate-800"
        title="board-tile"
      />
      <div
        class="flex flex-wrap justify-center content-center w-1/3 h-1/3 text-white bg-red-400"
        title="board-tile"
      />
      <div
        class="flex flex-wrap justify-center content-center w-1/3 h-1/3 text-white bg-slate-800"
        title="board-tile"
      />
      <div
        class="flex flex-wrap justify-center content-center w-1/3 h-1/3 text-white bg-red-400"
        title="board-tile"
      />
      <div
        class="flex flex-wrap justify-center content-center w-1/3 h-1/3 text-white bg-slate-800"
        title="board-tile"
      />
      <div
        class="flex flex-wrap justify-center content-center w-1/3 h-1/3 text-white bg-red-400"
        title="board-tile"
      />
      <div
        class="flex flex-wrap justify-center content-center w-1/3 h-1/3 text-white bg-slate-800"
        title="board-tile"
      />
    </div>
  </div>
</div>
`;
